vllm/config.py

```diff
@@ -271,12 +278,28 @@ def _init_pooler_config(
         pooling_returned_token_ids: Optional[List[int]] = None
     ) -> Optional["PoolerConfig"]:
         if self.task == "embedding":
-            return PoolerConfig(
-                pooling_type=pooling_type,
-                pooling_norm=pooling_norm,
-                pooling_softmax=pooling_softmax,
-                pooling_step_tag_id=pooling_step_tag_id,
-                pooling_returned_token_ids=pooling_returned_token_ids)
+            pooling_config = get_pooling_config(self.model, self.revision)
+            if pooling_config is not None:
+                pooling_type_from_file = pooling_config["pooling_type"]
+                normalize_from_file = pooling_config["normalize"]
+                pooling_config_from_file = PoolerConfig(
+                    pooling_type=pooling_type_from_file,
+                    pooling_norm=normalize_from_file,
+                    pooling_softmax=pooling_softmax,
+                    pooling_step_tag_id=pooling_step_tag_id,
+                    pooling_returned_token_ids=pooling_returned_token_ids)
+                if pooling_type is not None:
+                    pooling_config_from_file.pooling_type = pooling_type
+                if pooling_norm is not None:
+                    pooling_config_from_file.pooling_norm = pooling_norm
```

Comment:
Can we clean this logic up a bit? In particular, we can directly assign the contents of `pooling_config` to the input variables before constructing `PoolerConfig`.

Reply:
Sure, let me know if I can improve more!